### ISSUE TITLE
Offer: Weapon Designer

### ISSUE THREAD

**Issue by vinurd** [TheGameCreators/GameGuruRepo#3621]
![weapons_construct1](https://user-images.githubusercontent.com/32761694/185282539-88623216-ecfe-4edc-889f-a52204607e48.png)
![weapon2](https://user-images.githubusercontent.com/32761694/185282559-07b71be8-b0ad-4d93-9a24-31c6842df2ea.png)
Good afternoon. A proposal to expand the arsenal of weapons with minimal casualties.
You will only need a few animations for the hands For different withdrawal of the magazine from the weapon. 1.Remove the ammo magazin.2 unhook the box, pull out the drum, insert the cartridges. The weapon itself can be made a modular constructor, where the handle is the main arm, we make up the body from it, you can assemble it from different blocks, then attach an animated ejection for the cartridge-reacting to the shot of the weapon. The trunk itself (can be a set)weapons, optics and various decorations, tubes and elements. Depending on the type of store installed, an animation will be triggered. The store can also be modified and assembled from small parts. Or you can download the model of the main body and the store.But the pen and the store should be put so the animation is tied to it. The constructor can be expanded and the thing will turn out to be completely unique, as well as the number of weapons that can be collected. The only thing is to take out all the weapon settings from the side. The strength of the weapon- breaks, jams. Rate of fire, range, whether there are optics (choice), sight (choice) how many rounds are fired, accuracy, armor-piercing, damage, explosiveness, tracers, laser, particle, sound of firing.

**Comment by davetgc** [comment]
Hi, are you offering to develop this?

**Comment by vinurd** [comment]
yes, But that's just an idea.I'm a weak programmer. This would become a full-fledged DLC for creating all kinds of weapons. There, you can just add only animations of holding the handle and animations of the  hand for recharging, and animated details. Plus, these additions can be made by packs to this weapon designer.
